persist-cli/src/main/java/io/ballerina/persist/cmd/Init.java

```diff
@@ -52,46 +60,67 @@ public class Init implements BLauncherCmd {
     private final PrintStream outStream = System.out;
     private final String configPath = PersistToolsConstants.CONFIG_SCRIPT_FILE;
 
-    private String name = "";
+    private String configName = "";
+    private String persistName = "";
     public String sourcePath = "";
     private static final String COMMAND_IDENTIFIER = "persist-init";
 
+
     Project balProject;
 
     @CommandLine.Option(names = {"-h", "--help"}, hidden = true)
     private boolean helpFlag;
 
     @Override
     public void execute() {
+        Path persistTomlPath = Paths.get(this.sourcePath, KEYWORD_PERSIST , PERSISTTOML);
+        Path databaseConfigPath = Paths.get(this.sourcePath, KEYWORD_MODULES, KEYWORD_CLIENTS,
+                DATABASE_CONFIGURATION_BAL);
         if (helpFlag) {
             String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(COMMAND_IDENTIFIER);
             errStream.println(commandUsageInfo);
             return;
         }
+        Generate persistCmd;
+        try {
+            Class<?> persistClass = Class.forName("io.ballerina.persist.cmd.Generate");
+            persistCmd = (Generate) persistClass.getDeclaredConstructor().newInstance();
+            persistCmd.setSourcePath(Paths.get(sourcePath).toAbsolutePath().toString());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
         try  {
```

Comment:
Why generate is called in init? Seems it is being called always

Reply:
As per the discussion, we decided to call the generate function in the init command to generate the clients inside the modules directory. Additionally, by using this method, we don't need to repeat the same validations again inside the Init command.